src/main/java/com/zimbra/cert/GetDomainCert.java

```diff
@@ -0,0 +1,106 @@
+package com.zimbra.cert;
+
+import com.zimbra.common.account.Key.DomainBy;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.soap.AdminConstants;
+import com.zimbra.common.soap.CertMgrConstants;
+import com.zimbra.common.soap.Element;
+import com.zimbra.common.util.ZimbraLog;
+import com.zimbra.cs.account.Domain;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.account.accesscontrol.Rights.Admin;
+import com.zimbra.cs.service.admin.AdminDocumentHandler;
+import com.zimbra.soap.ZimbraSoapContext;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Admin Handler class to get information about a domain certificate. It provides a way to access
+ * all needed attributes of X.509 certificate using a standard java.security.cert package.
+ */
+public class GetDomainCert extends AdminDocumentHandler {
+
+  private static final String CERT_TYPE = "X.509";
+  private static final String DATE_PATTERN = "MMM dd yyyy HH:mm:ss z";
+
+  /**
+   * Handles the request. Searches a domain by name, checks admin rights (accessible to global and
+   * delegated admin of requested domain), decrypts X.509 certificate, creates response element.
+   *
+   * @param request {@link Element} representation of {@link
+   *     com.zimbra.soap.admin.message.GetDomainCertRequest}
+   * @param context request context
+   * @return {@link Element} representation of {@link
+   *     com.zimbra.soap.admin.message.GetDomainCertResponse}
+   * @throws ServiceException in case if a requested domain could not be found or if an error occurs
+   *     during certificate parsing.
+   */
+  @Override
+  public Element handle(Element request, Map<String, Object> context) throws ServiceException {
+
+    ZimbraSoapContext zsc = getZimbraSoapContext(context);
+
+    Provisioning prov = Provisioning.getInstance();
+    String domainName = request.getAttribute(AdminConstants.A_DOMAIN);
+    Domain domain = prov.get(DomainBy.name, domainName);
```

Comment:
To keep the APIs(GetCert and this one) consistent, consider adding the ability to accept requests by domain ID as well. We may accomplish this by making the **name** parameter optional and the **id** parameter obligatory :)

Reply:
i would modify by id then only like it is for a server
i've checked admin panel 
before calling this api they would already have the id information 
no need for additional option by name then